multidict/_multidict_iter.c

```diff
@@ -0,0 +1,309 @@
+    0,                                             /* tp_itemsize */
+    (destructor)multidict_iter_dealloc,            /* tp_dealloc */
+    0,                                             /* tp_print */
+    0,                                             /* tp_getattr */
+    0,                                             /* tp_setattr */
+    0,                                             /* tp_reserved */
+    0,                                             /* tp_repr */
+    0,                                             /* tp_as_number */
+    0,                                             /* tp_as_sequence */
+    0,                                             /* tp_as_mapping */
+    0,                                             /* tp_hash */
+    0,                                             /* tp_call */
+    0,                                             /* tp_str */
+    0,                                             /* tp_getattro */
+    0,                                             /* tp_setattro */
+    0,                                             /* tp_as_buffer */
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC,       /* tp_flags */
+    0,                                             /* tp_doc */
+    (traverseproc)multidict_iter_traverse,         /* tp_traverse */
+    0,                                             /* tp_clear */
+    0,                                             /* tp_richcompare */
+    0,                                             /* tp_weaklistoffset */
+    PyObject_SelfIter,                             /* tp_iter */
+    (iternextfunc)multidict_keys_iter_iternext,    /* tp_iternext */
+    0,                                             /* tp_methods */
+    0,                                             /* tp_members */
+    0,                                             /* tp_getset */
+    0,                                             /* tp_base */
+    0,                                             /* tp_dict */
+    0,                                             /* tp_descr_get */
+    0,                                             /* tp_descr_set */
+    0,                                             /* tp_dictoffset */
+    0,                                             /* tp_init */
+    0,                                             /* tp_alloc */
+    (newfunc)multidict_iter_new,                   /* tp_new */
+    0,
+};
+
+static struct PyModuleDef _multidict_iter_module = {
+    PyModuleDef_HEAD_INIT,       /* m_base */
+    "multidict._multidict_iter", /* m_name */
+    NULL,                        /* m_doc */
+    -1,                          /* m_size */
+    NULL,                        /* m_methods */
+    NULL,                        /* m_reload */
+    NULL,                        /* m_traverse */
+    NULL,                        /* m_clear */
+    NULL                         /* m_free */
+};
+
+PyMODINIT_FUNC
+PyInit__multidict_iter(void)
```

Comment:
We don't need a module for iterators. I'm ok with putting them into a separate `.c` file but no python module.
Right now `pair_list.c` intentionally is not a module too but a file with `init` function.
Later we'll convert it into `_multidict` C extention which is built from several .c files but is a single python module.

Reply:
Ok